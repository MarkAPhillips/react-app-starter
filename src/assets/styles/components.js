import styled from 'styled-components';

export const Main = styled.main`
    padding: 1em;
`;

export const Header = styled.h1`
    font-size: 1.5em;
`;

export const Section = styled.section``;

export const Button = styled.button``;

export const Input = styled.input``;
