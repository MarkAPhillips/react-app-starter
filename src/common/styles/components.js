import styled from 'styled-components';

export const Main = styled.main`
    padding: 1em;
`;

export const Header = styled.h1`
    font-size: 1.5em;
`;

export const Section = styled.section``;
