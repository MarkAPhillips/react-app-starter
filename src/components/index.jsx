import React from 'react';
import styled from 'styled-components';

const MainPanel = styled.main``;

const App = () => <MainPanel />;

export default App;
