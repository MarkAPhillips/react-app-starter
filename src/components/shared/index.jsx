export * from './CheckBox';
export * from './Icon';
