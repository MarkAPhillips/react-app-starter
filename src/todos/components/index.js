export { TodoAddItem } from './TodoAddItem';
export { TodoList } from './TodoList';
export { TodoListItem } from './TodoListItem';
