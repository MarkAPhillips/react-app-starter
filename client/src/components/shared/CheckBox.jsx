import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { TodoItemText, CheckBoxLabel, CheckBoxPanel } from './styles';

const defaultProps = {
  onChange: _.noop,
};

const propTypes = {
  onChange: PropTypes.func,
  id: PropTypes.string.isRequired,
  completed: PropTypes.bool.isRequired,
  item: PropTypes.string.isRequired,
};

export const CheckBox = ({
  item, completed, onChange, id,
}) => (
  <CheckBoxPanel>
    <input
      type="checkbox"
      name="chkBox"
      defaultChecked={completed}
      onChange={onChange}
      id={id}
    />
    <CheckBoxLabel htmlFor="chkBox">
      <TodoItemText completed={completed}>{item}</TodoItemText>
    </CheckBoxLabel>
  </CheckBoxPanel>
);

CheckBox.propTypes = propTypes;
CheckBox.defaultProps = defaultProps;
