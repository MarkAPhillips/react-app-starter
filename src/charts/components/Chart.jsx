import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { lifecycle, compose, withHandlers } from 'recompose';
import { ChartPanel, ChartSvg } from '../styles';
import build from '../utils/chartbuilder';
import { onFetch, getCatalogueDataForChart } from '../redux/';

const propTypes = {
  onRef: PropTypes.func.isRequired,
};

const Chart = ({ onRef }) => (
  <ChartPanel>
    <ChartSvg innerRef={onRef} />
  </ChartPanel>
);

const mapStateToProps = state => ({
  data: getCatalogueDataForChart(state),
});

Chart.propTypes = propTypes;

const enhance = compose(
  connect(mapStateToProps),
  withHandlers(() => {
    let elem = null;
    return {
      onRef: () => (ref) => {
        elem = ref;
      },
      drawChart: () => data => build(elem, data),
    };
  }),
  lifecycle({
    componentWillMount() {
      this.props.dispatch(onFetch());
    },
    componentWillReceiveProps(nextProps) {
      this.props.drawChart(nextProps.data);
    },
  }),
);

export default enhance(Chart);

