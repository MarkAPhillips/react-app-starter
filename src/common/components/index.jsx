import React from 'react';
import styled from 'styled-components';
import Chart from '../../charts/components/Chart';

const MainPanel = styled.main``;

const App = () => <MainPanel><Chart /></MainPanel>;

export default App;
