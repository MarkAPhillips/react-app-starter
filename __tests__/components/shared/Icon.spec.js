import React from 'react';
import { componentSnapshotShouldMatch } from '../../testUtils';
import { Icon } from '../../../src/components/shared';

describe('Icon specs', () => {
  it('shoud render the component', () => {
    const component = <Icon iconClassName="fa fa-trash" />;
    componentSnapshotShouldMatch(component);
  });
});
