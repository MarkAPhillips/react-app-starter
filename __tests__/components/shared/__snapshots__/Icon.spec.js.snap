// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon specs shoud render the component 1`] = `
<i
  aria-hidden="true"
  className="fa fa-trash"
/>
`;
